File: Upcoder/Lập trình cơ bản/2.Lệnh rẽ nhánh/GIO_PHUT_GIAY/GIO_PHUT_GIAY.cs
using System;

namespace giophucgiay
{
    public class Program
    {
        public static void Main(string[] args)
        {
            ushort gio, phut, giay;
            string[] input = Console.ReadLine().Split(' ');

            gio = ushort.Parse(input[0]);
            phut = ushort.Parse(input[1]);
            giay = ushort.Parse(input[2]);

            if (gio > 23 || phut > 59 || giay > 59)
            {
                Console.Write("NO");
            }
            else
            {
                Console.Write("YES");
                if (giay == 59)
                {
                    phut++;
                    giay = 0;
                }
                else
                {
                    giay++;
                }

                if (phut > 59)
                {
                    gio++;
                    phut = 0;
                    giay = 0;
                }

                if (gio > 23)
                {
                    gio = 0;
                    phut = 0;
                    giay = 0;
                }

                Console.WriteLine($"{gio}:{phut}:{giay}");
            }
        }
    }
}

File: Upcoder/Lập trình cơ bản/2.Lệnh rẽ nhánh/GIO_PHUT_GIAY/GIO_PHUT_GIAY.py
# Nhập giờ, phút, giây
gio, phut, giay = map(int, input().split())

# Kiểm tra giá trị hợp lệ
if gio > 23 or phut > 59 or giay > 59:
    print("NO")
else:
    print("YES")
    
    # Tăng giây
    if giay == 59:
        phut += 1
        giay = 0
    else:
        giay += 1

    # Kiểm tra phút
    if phut > 59:
        gio += 1
        phut = 0
        giay = 0

    # Kiểm tra giờ
    if gio > 23:
        gio = 0
        phut = 0
        giay = 0

    # Xuất giờ, phút, giây
    print("{}:{}:{}".format(gio, phut, giay), end="")
